src/foam/core/ImportsExports.js

```diff
@@ -163,34 +163,45 @@ foam.CLASS({
   ],
 
   methods: [
+    function getExportMap() {
+      var m = {};
+      var bs = this.cls_.getAxiomsByClass(foam.core.Export);
+      for ( var i = 0 ; i < bs.length ; i++ ) {
+        var b = bs[i];
+
+        if ( b.key ) {
+          var path = b.key.split('.');
+
+          var a = this.cls_.getAxiomByName(path[0]);
+
+          foam.assert(!!a, 'Unknown axiom: "', path[0], '" in model: ',
+                      this.cls_.id, ", trying to export: '", b.key, "'");
+
+          // Axioms have an option of wrapping a value for export.
+          // This could be used to bind a method to 'this', for example.
+          var e = a.exportAs ? a.exportAs(this, path.slice(1)) : this[path[0]];
```

Comment:
Yes. I remember that now. It is still inconsistent though that 

`foo` => `foo` (value), `foo$` (slot)
and
`foo.bar` => `foo` (slot)

Isn't that what happens here?